### DISCUSSION TITLE
Problema configurazione Zwave JS ui

### DISCUSSION THREAD

**Discussion by pavlov82** [indomus/forum#2598] [ZigBee, Z-Wave, Thread, Matter]
Ciao a tutti, ieri stavo configurando il container docker per zwave seguendo la guida che avevate messo a disposizione sul sito.

Sono riuscito a instanziare il container docker, tramite docker compose, con questa configurazione:

```version: '3.9'
services:
  home-assistant:
    container_name: home-assistant
    image: ghcr.io/home-assistant/home-assistant:stable
    volumes:
    - /home/deb/homeassistant:/config
    - /run/dbus:/run/dbus:ro
    environment:
    - TZ=Europe/Rome
    network_mode: host
    restart: always
  cloudflared:
    container_name: cloudflared
    hostname: cloudflared
    image: cloudflare/cloudflared:latest
    restart: always
    volumes:
    - /home/deb/cloudflared:/home/nonroot/.cloudflared
    command: tunnel run f6b1d530-79d2-41c0-b4e0-12005a9c4610
  zwavejsui:
    container_name: zwavejsui
    image: zwavejs/zwave-js-ui:latest
    restart: unless-stopped
    devices:
      - "/dev/ttyUSB0:/dev/zwave"
    volumes:
      - "/home/deb/zwavejsui:/usr/src/app/store"
    ports:
      - "3000:3000"
      - "8091:8091"
```
Quando però cerco di configurare da interfaccia zwave js ui alla selezione della serial port ottengo questo errore:

"Driver: failed to open the serial port Error: no such file or directory, cannot open /dev/ttyUSB0".

Mi sapreste dire se sto sbagliando qualcosa o se mi riuscite ad aiutare a risolvere il problema?

Grazie mille.
Ciao Paolo.

**Comment by indomus** [comment]
Si suppone tu lo sia implementando su docker@debian.
da qui la domanda: hai configurato i dialout? https://indomus.it/guide/le-prime-cose-da-fare-dopo-aver-installato-linux-debian-12-per-la-propria-smart-home/#dialout

**Reply by pavlov82** [reply]
Ciao,
ero abbastanza convinto di averlo fatto, ne ho comunque approfittato e ho lanciato comunque di nuovo il comando per sicurezza (ancora me lo fossi perso da qualche parte).

Per cui, considerando che ho configurato l'utente deb, ho lanciato i comando:

`sudo usermod -aG dialout deb`

Ho rimosso i container, cancellato la folder con la config, ricreata come da guida, ho reistanziato il container.

Persiste pero l'errore: "Driver: failed to open the serial port Error: no such file or directory, cannot open /dev/ttyUSB0".

**Reply by snakuzzo** [reply]
Se hai questo errore vuol dire che la chiavetta non è montata su `/dev/ttyUSB0`

**Reply by pavlov82** [reply]
Era anche il mio dubbio iniziale però ho fatto anche il bash per le porte che ho trovato sempre nelle guide e mi ritorna questo:

```
/dev/bus/usb/001/001 - Linux_6.1.0-11-amd64_xhci-hcd_xHCI_Host_Controller_0000:00:14.0
/dev/bus/usb/001/006 - Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_34e89f245585ed11b04dd09f9d1cc348
/dev/gpiochip1 -
/dev/ttyUSB0 - Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_34e89f245585ed11b04dd09f9d1cc348
/dev/bus/usb/001/004 - Realtek_Bluetooth_Radio_00e04c000001
/dev/bus/usb/001/003 - 1a40_USB_2.0_Hub
/dev/bus/usb/002/001 - Linux_6.1.0-11-amd64_xhci-hcd_xHCI_Host_Controller_0000:00:14.0
```

Per cui la chiavetta secondo me è effettivamente su ttyUSB0

**Reply by snakuzzo** [reply]
Forse mi sono spiegato male.
L'errore te lo dà in fase di creazione del container o in fase di start del container ?

**Reply by pavlov82** [reply]
Vai tranquillo, tieni conto che ci sclero da ieri per cui ci sta che non abbia capito bene nemmeno io 🤣 

Il container viene su tranquillo e riesco anche ad accedere all'interfaccia web (infatti se sbaglio volutamente la porta, il container non viene nemmeno su).

Quando configuro da Settings dentro all'interfaccia si zwave js ui, li se metto come serial port /dev/ttyUSB0 mi da l'errore che ho indicato sopra.

Grazie mille ancora.
Ciao Paolo.

**Reply by snakuzzo** [reply]
Ma hai visto come hai mappato la porta nel docker-compose ?

```
    devices:
      - "/dev/ttyUSB0:/dev/zwave"
``` 

Secondo te come può mai funzionare se poi nella configurazione dell'applicazione scrivi `/dev/ttyUSB0` ?

Qui il problema è che hai seguito la guida senza capire cosa stavi facendo...hai semplicemente fatto copia-incolla dando tutto per scontato.
Quella roba lì vuol dire che `/dev/ttyUSB0` dell'host viene mappato con `/dev/zwave` del container. Quindi l'applicazione la vedrà come `/dev/zwave`

**Reply by pavlov82** [reply]
Yes, grazie funziona correttamente.

Avevo dato per scontato che fosse da inserire la ttyUSB0 dalla guida (errore mio).

Comunque le guide sono fatte benissimo, quindi continuate cosi: mi porto a casa di approfondire di piu la prossima volta prima di disturbarvi 😆 promesso.

Grazie ancora e continuate cosi.

Ciao Paolo.

**Reply by snakuzzo** [reply]
Ok, più che altro devi adottare l'approccio giusto che consiste nel capire cosa stai "incollando".
Se non sai cosa vuol dire quella roba, vai su google (o altro), cerchi e capisci...se dai tutto per scontato non imparerai mai nulla.
Le guide, per quanto cerchiamo di farle precise, non sono dei manuali di istruzioni

**Reply by pavlov82** [reply]
Hai assolutamente ragione e gia mi state tanto aiutando a imparare un sacco di cose sia con cloudflare sia con debian.

Devo ammettere che zwave l'ho preso un po troppo sottogamba... lesson learned 🥇 

Grazie ancora, scusate la niubbaggine e continuate cosi.

Ciao Paolo.
